### PR TITLE
add recommended Elixir and Erlang/OTP combinations to CI and migrate it from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by adrianomitre**
This pull request migrate the CI from Travis CI to GitHub Actions, as requested by José Valim in [this comment](https://github.com/elixir-plug/plug/pull/957#issuecomment-647738665).

In addition, it makes sure all recommended Elixir and Erlang/OTP combinations are run on CI.

Quoting José Valim:
> I would still advise folks to go with:
> * For each Elixir version, add a run with earliest supported Erlang/OTP
> * For the last Elixir version, also test the latest supported Erlang/OTP

Please refer to the original discussion on https://github.com/dashbitco/broadway/pull/188 for the full rationale.

Please refer to [Yet another Elixir and Erlang docker image - Chat / Discussions - Elixir Forum](https://elixirforum.com/t/yet-another-elixir-and-erlang-docker-image/28740) for the rationale behing Elixir images by [Hexpm's Bob](https://github.com/hexpm/bob#docker-images).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          mix local.hex --force
+          apk add --no-progress --update git bash wget nodejs npm
+          wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+          export NVM_DIR="$HOME/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+          nvm install 12.16 && nvm use 12.16
+          mix deps.get
+      - run: mix test
+      - run: cd installer && elixir -S mix test
+      - run: cd assets && npm install && npm test
```
Comment (josevalim) [inline .github/workflows/ci.yml:35]: We should probably run the npm tests in a separate task. The same for the npm setup. See configuration here:

https://github.com/phoenixframework/phoenix/pull/3557/files#diff-e9f950f17198d3d5e3122a44230a09b9R78

Comment (adrianomitre) [reply .github/workflows/ci.yml]: Done in https://github.com/phoenixframework/phoenix/pull/3944/commits/22cf1170867bc9c59a4f5ae5c26f0af4524e26e3

**Comment by adrianomitre** [comment]
Evidence of non-deterministic behaviour (same everything, only [an empty commit](https://github.com/phoenixframework/phoenix/pull/3944/commits/34b6eb1e831a03a0cdc6a97d7fe3fee199aae2a6) apart):
- https://github.com/phoenixframework/phoenix/pull/3944/checks?check_run_id=941932050 <- fail
- https://github.com/phoenixframework/phoenix/pull/3944/checks?check_run_id=942088825 <- pass

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      - uses: actions/checkout@v2.3.1
+      - name: Install Dependencies
+        run: |
+          mix local.rebar --force
+          mix local.hex --force
+          apk add --no-progress --update git bash wget nodejs npm
+          wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+          export NVM_DIR="$HOME/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+          nvm install 12.16 && nvm use 12.16
```
Comment (josevalim) [inline .github/workflows/ci.yml:31]: Do we need to run these instructions now that we no longer run NPM tests on this suite? :)

Comment (adrianomitre) [reply .github/workflows/ci.yml]: Good catch. Removed in 67e6dd86f7074c8dfff25b3562ed5f76db2fe639

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**